UI/win-dll-blocklist.c

```diff
@@ -0,0 +1,300 @@
+	{L"\\action_x64.dll", 0, 0, TS_IGNORE},
+
+	// ASUS GamerOSD, breaks DX11 things
+	{L"\\atkdx11disp.dll", 0, 0, TS_IGNORE},
+
+	// Malware
+	{L"\\sendori.dll", 0, 0, TS_IGNORE},
+
+	// Astril VPN Proxy, hooks stuff and crashes
+	{L"\\asproxy64.dll", 0, 0, TS_IGNORE},
+
+	// Nahimic Audio
+	{L"\\nahimicmsidevprops.dll", 0, 0, TS_IGNORE},
+	{L"\\nahimicmsiosd.dll", 0, 0, TS_IGNORE},
+
+	// FRAPS hook
+	{L"\\fraps64.dll", 0, 0, TS_IGNORE},
+
+	// ASUS GPU TWEAK II OSD
+	{L"\\gtii-osd64.dll", 0, 0, TS_IGNORE},
+	{L"\\gtii-osd64-vk.dll", 0, 0, TS_IGNORE},
+
+	// EVGA Precision, D3D crashes
+	{L"\\pxshw10_x64.dll", 0, 0, TS_IGNORE},
+
+	// Wacom / Other tablet driver, locks up UI
+	{L"\\wintab32.dll", 0, 0, TS_IGNORE},
+
+	// Adobe Dynamic Link (Adobe CC), crashes in its own thread
+	{L"\\mc_trans_video_imagescaler.dll", 0, 0, TS_IGNORE},
+
+	// Weird Polish banking "security" software, breaks UI
+	{L"\\wslbscr64.dll", 0, 0, TS_IGNORE},
+
+	// Various things hooking with EasyHook that probably shouldn't touch OBS
+	{L"\\easyhook64.dll", 0, 0, TS_IGNORE},
+
+	// Ultramon
+	{L"\\rtsultramonhook.dll", 0, 0, TS_IGNORE},
+
+	// HiAlgo Boost, locks up UI
+	{L"\\hookdll.dll", 0, 0, TS_IGNORE},
+
+	// Adobe Core Sync? Crashes NDI.
+	{L"\\coresync_x64.dll", 0, 0, TS_IGNORE},
+
+	// Fasso DRM, crashes D3D
+	{L"\\f_sps.dll", 0, 0, TS_IGNORE},
+
+	// Korean banking "security" software, crashes randomly
+	{L"\\t_prevent64.dll", 0, 0, TS_IGNORE},
+
+	// Generic named unity capture filter. Unfortunately only a forked version
+	// has a critical fix to prevent deadlocks during enumeration. We block
+	// all versions since if someone didn't change the DLL name they likely
+	// didn't implement the deadlock fix.
+	// Reference: https://github.com/schellingb/UnityCapture/commit/2eabf0f
+	{L"\\unitycapturefilter64bit.dll", 0, 0, TS_IGNORE},
+
+	// VSeeFace capture filter < v1.13.38b3 without above fix implemented
+	{L"\\vseefacecamera64bit.dll", 0, 1666993098, TS_LESS_THAN},
+};
+
+static bool is_module_blocked(wchar_t *dll, uint32_t timestamp)
+{
+	size_t len;
+	len = wcslen(dll);
+
+	wcslwr(dll);
+
+	bool should_block = false;
+
+	for (int i = 0; i < _countof(blocked_modules); i++) {
+		const blocked_module_t *b = &blocked_modules[i];
+		wchar_t *dll_ptr;
+
+		if (len >= b->name_len)
+			dll_ptr = dll + len - b->name_len;
+		else
+			dll_ptr = dll;
+
+		if (!wcscmp(dll_ptr, b->name)) {
+			if (b->method == TS_IGNORE)
+				return true;
+			else if (b->method == TS_EQUAL &&
+				 timestamp == b->timestamp)
+				return true;
+			else if (b->method == TS_LESS_THAN &&
+				 timestamp < b->timestamp)
+				return true;
+			else if (b->method == TS_GREATER_THAN &&
+				 timestamp > b->timestamp)
+				return true;
+			else if (b->method == TS_ALLOW_ONLY_THIS) {
+				should_block = true;
+
+				if (timestamp == b->timestamp)
+					return false;
+			}
+		}
+	}
+
+	return should_block;
+}
+
+NTSTATUS NtMapViewOfSection_hook(HANDLE SectionHandle, HANDLE ProcessHandle,
```

Comment:
Can we add `static` qualifier?
```suggestion
static NTSTATUS NtMapViewOfSection_hook(HANDLE SectionHandle, HANDLE ProcessHandle,
```